subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -188,11 +188,20 @@ static void reset_adv_link(void)
 
 	k_delayed_work_cancel(&link.prot_timer);
 
-	/* Clear everything except the retransmit and protocol timer
-	 * delayed work objects.
-	 */
-	(void)memset(&link, 0, offsetof(struct pb_adv, tx.retransmit));
-	link.rx.id = XACT_NVAL;
+	if (atomic_test_bit(link.flags, PROVISIONER)) {
+		/* Clear everything except the retransmit and protocol timer
+		 * delayed work objects.
+		 */
+		(void)memset(&link, 0, offsetof(struct pb_adv, tx.retransmit));
+		link.rx.id = XACT_NVAL;
+	} else {
+		/* Accept another provisioning attempt */
+		link.id = 0;
+		atomic_clear(link.flags);
+		link.rx.id = 0x7F;
+		link.tx.id = 0xFF;
```

Comment:
I'm starting to think we might want a define for this magic `0x7f` since it seems quite prevalent in the code.

Reply:
Agreed. I added a new commit with the define.